common/flatpak-run-wayland.c

```diff
@@ -57,6 +98,32 @@ static const struct wl_registry_listener registry_listener = {
   .global_remove = registry_handle_global_remove,
 };
 
+/* Similar to wl_display_connect(), but does not use WAYLAND_SOCKET,
+ * which can only be used once, and also does not unset environment
+ * variables, which would not be thread-safe. */
+static struct wl_display *
+connect_to_wayland_display (void)
+{
+  struct sockaddr_un sockaddr = {0};
+  g_autofree char *socket_path = NULL;
+  int fd = -1;
```

Comment:
Could use `g_autofd` and `g_steal_fd`.

Reply:
that would require glib 2.76 and meson.build currently requires 2.46. Also, there's no existing usage in this project.

Reply:
The libglnx submodule provides a backport of `g_steal_fd()`, so you can use that freely.

The libglnx submodule also provides `glnx_autofd`; please use that in preference to `g_autofd` for now (its meaning is basically the same).

Both are used in flatpak already, for example in `common/flatpak-run-dbus.c`.